heroku-24-build/Dockerfile

```diff
@@ -8,3 +8,5 @@ USER 1002
 ENV CNB_USER_ID=1002
 ENV CNB_GROUP_ID=1000
 ENV CNB_STACK_ID "heroku-24"
+ENV CNB_TARGET_OS=$TARGETOS
+ENV CNB_TARGET_ARCH=$TARGETARCH
```

Comment:
Once the upstream `lifecycle` bug is fixed and our builder updated to that version, will we remove these? I presume it will be safe to do so, since once `lifecycle` is setting them for us there won't be backwards compat concerns for removing an env var that was once set before?